src/pipecat/utils/wathcdog_async_iterator.py

```diff
@@ -0,0 +1,42 @@
+#
+# Copyright (c) 2024–2025, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+import asyncio
+from typing import TYPE_CHECKING, AsyncIterator, Optional
+
+if TYPE_CHECKING:
+    from pipecat.processors.frame_processor import FrameProcessor
+
+
+class WatchdogAsyncIterator:
+    def __init__(self, async_iterable, *, processor: "FrameProcessor", timeout: float = 1.0):
+        self._async_iterable = async_iterable
+        self._processor = processor
+        self._timeout = timeout
+        self._iter: Optional[AsyncIterator] = None
+
+    def __aiter__(self):
+        return self
+
+    async def __anext__(self):
+        if not self._iter:
+            self._iter = await self._ensure_async_iterator(self._async_iterable)
+
+        try:
+            item = await asyncio.wait_for(self._iter.__anext__(), timeout=self._timeout)
+            self._processor.reset_watchdog()
+            return item
+        except asyncio.TimeoutError:
+            self._processor.reset_watchdog()
+            return await self.__anext__()
+        except StopAsyncIteration:
+            raise
```

Comment:
Wouldn't be better to use a loop instead of recursion in this case ?

```
while True:
    try:
        item = await asyncio.wait_for(self._iter.__anext__(), timeout=self._timeout)
        self._processor.reset_watchdog()
        return item
    except asyncio.TimeoutError:
        self._processor.reset_watchdog()
        continue
    except StopAsyncIteration:
        raise
```

Reply:
Fixed!